src/librustc_mir/interpret/eval_context.rs

```diff
@@ -949,6 +949,26 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
     }
 }
 
+impl<'mir, 'tcx: 'mir, M: super::intern::CompileTimeMachine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
+    /// A helper function that allocates memory for the layout given and gives you access to mutate
+    /// it. Once your own mutation code is done, the backing `Allocation` is removed from the
+    /// current `Memory` and returned.
```

Comment:
Wow, this breaks through so many abstraction barriers... why?^^

Reply:
Seems like really you want this only for interning, so wouldn't it make more sense to also incorporate interning and move the entire thing to `intern.rs`? It's still somewhat unsettling but at least then it is less general-purpose, aka less dangerous.^^